src/Compilers/CSharp/Test/Semantic/Semantics/BetterCandidates.cs

```diff
@@ -81,6 +82,345 @@ class B {}
             CompileAndVerify(compilation, expectedOutput: "2");
         }
 
+        //When a method group contains both instance and static members, we discard the static members if invoked with an instance receiver.
+        [Fact]
+        public void TestInstanceReceiver02()
+        {
+            var source =
+@"class Program
+{
+    public static void Main()
+    {
+        Program p = new Program();
+        p.Main2();
+    }
+    void Main2()
+    {
+        this.M(null);
+    }
+    public static void M(A a) { System.Console.WriteLine(1); }
+    public void M(B b) { System.Console.WriteLine(2); }
+}
+class A {}
+class B {}
+";
+            CreateCompilationWithoutBetterCandidates(source, options: TestOptions.ReleaseExe).VerifyDiagnostics(
+                // (10,14): error CS0121: The call is ambiguous between the following methods or properties: 'Program.M(A)' and 'Program.M(B)'
+                //         this.M(null);
+                Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("Program.M(A)", "Program.M(B)").WithLocation(10, 14)
+                );
+            var compilation = CreateCompilationWithBetterCandidates(source, options: TestOptions.ReleaseExe).VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput: "2");
```

Comment:
Would it make sense to check the semantic model and checking candidate symbols? I'm assuming the static overload is not listed.